src/analysisd/rules.c

```diff
@@ -2052,7 +2052,7 @@ STATIC char *loadmemory(char *at, const char *str, OSList* log_msg)
                 merror(MEM_ERROR, errno, strerror(errno));
                 return (NULL);
             }
-            strncpy(at, str, strsize);
+            memcpy(at, str, strsize);
```

Comment:
Not sure about this but doesn't this skip adding the null terminator that `strncpy` would do? I think you would have to pass `strsize + 1` so that it also copies the null terminator.

Reply:
Hi @RamiroRD 

The NUL- terminated string is guaranteed because calloc allocates memory and initializes it to 0. the element `strsize + 1` is already 0.

Best regards
Hanes